tests/tst_ncio.F90

```diff
@@ -1,11 +1,10 @@
-program test_ncio
-
-! test for NCEPLIBS-ncio
+! This is a test for the NCEPLIBS-ncio package.
```

Comment:
Is there a reason to invoke the preprocessor on this code?
i.e. why `.F90`?

Reply:
By default I believe all files should be F90.

Reply:
Only when a pre-processor (`-fpp`) needs to be invoked to generate the `.f90`, which is what the compiler will compile.
In other words, in this case, the pre-processor will take `.F90` and preprocess it into `.f90`. It is the `.f90` that is compiled.
The `.F90` and `.f90` in this case are identical, since there are no `ifdef`'s for e.g. in this code.

Reply:
Yes, I quite understand. Using F90 everywhere always works, whereas using f90 breaks when someone adds use of pre-processor.

So using F90 is safer and what we do in netcdf-fortran. In this case, pre-processor symbols are always available, and when writing portable code, are often needed.

Reply:
Well, I have several comments on that:
- If everything uses `.F90` than cpp will be run on all files, regardless of whether there are cpp macros or not. Not very efficient.
- using `.F90` for files that really need cpp, serves as a documentation and reminds people there are some cpp inside. Which i think is useful.
- When someone adds a macro to `.f90`, they should rename the file to `.F90`. It's simple as that.
- Littering code with `ifdef` everywhere should be discouraged.  Non-portable code should be located in only few files, not added everywhere.

Reply:
OK we will agree to disagree and I will be making all NCEPLIBS files .F90, based on my experiences with maintaining fortran on PIO and netcdf-fortran.